docs/earthfile/earthfile.md

```diff
@@ -908,6 +908,15 @@ RUN git fetch --unshallow
 ```
 {% endhint %}
 
+{% hint style='warning' %}
+As of Earthly v0.7.21, git credentials are no longer stored in the `.git/config` file; this includes the username.
+This means any ssh-based or https-based fetches or pushes will no longer work unless you restore the configured url,
+which can be done with:
+```Dockerfile
+RUN git remote set-url origin <git-url>
+```
+{% endhint %}
```

Comment:
Can you also update the example above this hint? It includes a `git fetch --unshallow`, which no longer works as is.

Reply:
i was on the fence about adding this since it's only required if you're using authentication, but I added a comment for that.

Reply:
Ah ok :+1: